packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -11,19 +11,29 @@ interface ComboboxContentProps {
   readonly options: string[];
 
   /**
-   * onSelection handler.
+   * onSelect handler.
    */
-  readonly onSelection: (selection: string) => void;
+  readonly onSelect: (selection: string) => void;
 
   /**
    * actions
    */
   readonly children: ReactElement | ReactElement[];
+
+  /**
+   * pre selected option
+   * @default ""
+   * @optional
+   * @type string
+   * */
+  readonly selected?: string;
 }
 
 export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
   const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
+  const [selectedOption, setSelectedOption] = useState<string>(
+    props.selected || "",
```

Comment:
Actually, now that I see it, do we need to track the state internally?

Reply:
technically no. for single select you're right, we could get away with not having an internal state because you click, it selects, closes and you're done

however for multi select since you'll be able to click on something, click on something else, maybe change your mind and undo one of them etc. there's all sorts of changes happening to the selections that we would not want to inform the consumer of until you are done and happy with your choices. otherwise that could needlessly trigger all sorts of updates to anything listening to your selections

so we'll eventually need this because the internal and external states will have valid cases where they are temporarily out of sync

Reply:
though now that I say it, I feel like there could be cases where you do want to immediately see the selections happening live

almost feels like that could be a prop specifically for multi select. should it trigger `onSelect` the second you interact with options, or should it wait until you've closed it

Reply:
I believe most of the cases would be instant. If it does get to this point

> otherwise that could needlessly trigger all sorts of updates to anything listening to your selections

That's the throttling pattern where the consumer can also implement and that we don't have to worry about it until we hear otherwise.

```tsx
<MyComponent value={value} onChange={handleChange} />

...
const throttleFetch = throttle(doSomething, 300);

function handleChange(newValue) {
  // cancel existing func
  doSomething.cancel()

  // set new values
  setValue(newValue);

  // try again
  doSomething();
}
```

Our components need to be predictable and part of it being predictable is not handling a state that the user can change within it since you risk the internal state and external state being out-of-sync.

You also have the part where you have outside forces that can change the selected value on that component. That means you would have to sync the state internally. One example of that is saved filters that can be applied anytime. Not just on mount.

Plus, single select deffo don't need this so let's start with simple

Reply:
I'm not convinced a `throttle` or `debounce` works there though

like think about how people use these types of lists. I am assigning team members, and doing _something_ expensive with those selections, so I want to have a high degree of certainty that I have the final results before calculating based on the selections

if I have even 15 entires in the list, I start out and I pick Adam. next, I want to pick Zak. how long does it take me to either scroll down to the bottom of the list, or to use the search and type it in and then select that user?

I'd wager it's going to be more than 300ms, so you're going to have to bump that up significantly. I don't even know what value you'd use because sometimes if all your selections are beside one another, 300ms is fine - but if you need to spend time getting to your next selection it totally depends on how well the user can navigate our UX

if you make that debounce too high, it's going to feel laggy or unresponsive for those quick cases, and you can't guarantee it'll even be enough for users who make selections very slowly

that's where I'm like ok, there's no way we can know - we should offer the options to just wait until it closes. I do understand your argument about consistency and predictability but I don't see how someone could make a good UX using this component in this scenario (btw it is a real scenario I know that is coming soon, not just something I've made up!)

Reply:
That's a good point! 

> like think about how people use these types of lists. I am assigning team members, and doing something expensive with those selections, so I want to have a high degree of certainty that I have the final results before calculating based on the selections

Doesn't this land on a "Do what you need, double check the other _fields_, then click save when you're done" pattern?

I do get where you're coming from tho. We need to ensure we only support and build UX's that we understand and would do right now. Majority of the use cases in Jobber where you select things are still all under 1 big save button (settings, edit pages).

I don't doubt that last scenario tho. It will happen for sure. When it does, we should support that at that point since it comes with a lot of micro-interaction that other (majority of) use cases wouldn't.

Reply:
>Doesn't this land on a "Do what you need, double check the other fields, then click save when you're done" pattern?

typically yes, you're right. the case I have in mind is atypical in that it is doing live, async stuff to update your UI and generate further options that you would then choose, make sure you're happy with and then save. so it might be a complete outlier that we do not want to design for. also this is coming up relatively soon as I understand it, which is why I'd like to do some thinking about it now rather than waiting because the solution doesn't feel obvious to me

it feels like we're in a bit of a pickle with two conflicting constraints. we want these flows to feel familiar so that users learn it once and know it for all of the product. however; in this instance the cost is too great to have that same behavior so we need to do something different

Reply:
Agreed! Which is also a design conundrum! Since consistency should also be their responsibility.